src/entities/User/routes.ts

```diff
@@ -0,0 +1,125 @@
+import { WithAuth, auth } from 'decentraland-gatsby/dist/entities/Auth/middleware'
+import handleAPI from 'decentraland-gatsby/dist/entities/Route/handle'
+import routes from 'decentraland-gatsby/dist/entities/Route/routes'
+import { hashMessage, recoverAddress } from 'ethers/lib/utils'
+import { Request, Response } from 'express'
+import isEthereumAddress from 'validator/lib/isEthereumAddress'
+
+import { DiscourseService } from '../../services/DiscourseService'
+import { ErrorService } from '../../services/ErrorService'
+import isDebugAddress from '../Debug/isDebugAddress'
+
+import { GATSBY_DISCOURSE_CONNECT_THREAD, MESSAGE_TIMEOUT_TIME } from './constants'
+import UserModel from './model'
+import { ValidationMessage } from './types'
+
+export default routes((route) => {
+  const withAuth = auth()
+  route.get('/user/:address/is-validated', handleAPI(isValidated))
+  route.get('/user/validate', withAuth, handleAPI(getValidationMessage))
+  route.post('/user/validate', withAuth, handleAPI(checkValidationMessage))
+  route.delete('/user/validate', withAuth, handleAPI(deleteValidation))
+})
+
+const VALIDATIONS_IN_PROGRESS: Record<string, ValidationMessage> = {}
+
+function clearValidationInProgress(user: string) {
+  const validation = VALIDATIONS_IN_PROGRESS[user]
+  if (validation) {
+    clearTimeout(validation.messageTimeout)
+    delete VALIDATIONS_IN_PROGRESS[user]
+  }
+}
+
+function formatValidationMessage(address: string, timestamp: string) {
+  return `By signing and posting this message I'm linking my Decentraland DAO account ${address} with this Discourse forum account\n\nDate: ${timestamp}`
+}
+
+async function getValidationMessage(req: WithAuth) {
+  const address = req.auth!
+  const message = {
+    address,
+    timestamp: new Date().toISOString(),
+  }
+
+  const messageTimeout = setTimeout(() => {
+    delete VALIDATIONS_IN_PROGRESS[address]
+  }, MESSAGE_TIMEOUT_TIME)
+
+  VALIDATIONS_IN_PROGRESS[address] = {
+    ...message,
+    messageTimeout,
+  }
+
+  return formatValidationMessage(address, message.timestamp)
+}
+
+async function checkValidationMessage(req: WithAuth) {
+  const user = req.auth!
+  try {
+    const messageProperties = VALIDATIONS_IN_PROGRESS[user]
+    if (!messageProperties) {
+      throw new Error('Validation timed out')
+    }
+
+    const { address, timestamp } = messageProperties
+
+    const comments = await DiscourseService.fetchAllComments(Number(GATSBY_DISCOURSE_CONNECT_THREAD))
+    const timeWindow = new Date(new Date().getTime() - MESSAGE_TIMEOUT_TIME)
+
+    const filteredComments = comments.filter((comment) => new Date(comment.created_at) > timeWindow)
+    const validComment = filteredComments.find((comment) => {
```

Comment:
You could extract this bit to another function:

```typescript
 const timeWindow = new Date(new Date().getTime() - MESSAGE_TIMEOUT_TIME)

    const filteredComments = comments.filter((comment) => new Date(comment.created_at) > timeWindow)
    const validComment = filteredComments.find((comment) => {
      const addressRegex = new RegExp(address, 'i')
      const dateRegex = new RegExp(timestamp, 'i')

      return addressRegex.test(comment.cooked) && dateRegex.test(comment.cooked)
    })
```
    
Something like

```typescript
const validationComment = getValidationComment(comments, address, timestamp);
```

We can later put this in a UserService... 👀